### ISSUE TITLE
Ignores updated/new displayname files

### ISSUE THREAD

**Issue by cfenollosa** [pimutils/todoman#388]
*(Edit: The original title was "Ignores displayname files", and with that assumption I wrote this first description. After further research on the issue, I've updated it to reflect the new findings described on the next comment)*

Hello,

**Summary**

According to [the documentation](https://todoman.readthedocs.io/en/stable/configure.html#color-and-displayname), todoman searches for a file named `displayname` in every directory name, and uses its raw contents as the list name. If not found, todoman uses the folder name as list name.

**Version**

In my installation (3.5.0 installed from packages on Debian 10) this behavior is not happening.

**Configuration**

I have two folders (i.e. two task lists), and on each of them there is a file named `displayname`.
The files can be read and the content is a string containing the list name.

```sh
~$ cat /home/[username]/.local/share/vdirsyncer/storage/cd77043e-[redacted]/displayname
Tasks
~$ cat /home/cfenollosa/.local/share/vdirsyncer/storage/A0EE9E61-[redacted]/displayname 
Futuro
```

This is my config file:

```sh
~$ cat .config/todoman/todoman.conf 
[main]
path = ~/.local/share/vdirsyncer/storage/*
default_list = Tasks
default_due = 0
humanize = True
```

As you can see, my default list is "Tasks", i.e., folder `cd77043e-[redacted]`

I have also confirmed that my version contains the code to read the displayname:

```sh
~$ grep -R "displayname" /usr/lib/python3/dist-packages/todoman/
/usr/lib/python3/dist-packages/todoman/model.py:            with open(os.path.join(path, 'displayname')) as f:
Binary file /usr/lib/python3/dist-packages/todoman/__pycache__/model.cpython-37.pyc matches
```

Just in case, strace confirms that the displayname files are read

```sh
~$ strace todoman 2>&1 | grep displayname
openat(AT_FDCWD, "/home/[username]/.local/share/vdirsyncer/storage/cd77043e-[redacted]/displayname", O_RDONLY|O_CLOEXEC) = 4
openat(AT_FDCWD, "/home/[username]/.local/share/vdirsyncer/storage/A0EE9E61-[redacted]/displayname", O_RDONLY|O_CLOEXEC) = 4
```

**Behavior**

todoman is not using the content of these files when displaying tasks

```sh
~$ todoman
42  [ ]                   Task one @cd77043e-[redacted]                                                           
10  [ ]                   Task two @cd77043e-[redacted]

(more output follows, snipped)
```

todoman is not using the displayname when adding tasks, either.

Note that I have configured a default list by using their displayname.

```sh
~$ todoman new test
Usage: todoman new [OPTIONS] [SUMMARY]...

Error: Invalid value for "--list" / "-l": Tasks. Available lists are: cd77043e-[redacted], A0EE9E61-[redacted]
```

When using the folder name as list name, todoman works as expected

```sh
~$ todoman new test --list cd77043e-[redacted]
47  [ ]      test @cd77043e-[redacted]
```

**Conclusion**

Is there something wrong in my setup or is this maybe a bug?

Thanks!

**Comment by cfenollosa** [comment]
Okay, I did some more research and found the culprit and the solution.

After reading todoman source code, I found a suspicious keyword: `cache`. Now it all makes sense.

Leaving a comment here so other people can find it if they ever encounter the problem.

**Problem**

The initial run of `todoman` was executed when the task folders did NOT have a `displayname` file. This is because `vdirsyncer metasync` did not run and therefore didn't create this file.

The displayname was added later, manually.

Since todoman uses a cache, that cache is initialized but never updated whenever the `displayname` or `color` files appear.

**Solution**

Delete the cache and run `todoman` again

```sh
~$ mv ~/.cache/todoman/cache.sqlite3{,.bak}
~$ todoman
```

**Further work**

I suggest that `todoman` updates their cache if it finds the `displayname` or `color` files anew, or whenever the content of these files changes.

I'm leaving the issue opened so that an admin can read this last comment, and decide whether it would be a good idea to update the caches.

Admins: Feel free to close this issue. Thanks for reading.
